### PR TITLE
[nav] Removal of Radio Repeater code in jetson/nav

### LINKED ISSUE #869: [nav] Remove Radio repeater code
1/15 integration test revealed that the RR code in `jetson/nav` was causing segfaults in communicating with `base_station`. Making an issue to go along with future PR.

### PR DESCRIPTION
**Pull request by AjaySum**
resolves #869

Removal of all Radio Repeater code in `jetson/nav` to reflect current status of rover. This also fixes the former segmentation fault issue when working with `base_station/gui`.

### CODE REVIEW

File: jetson/nav/rover.cpp [hunk 1]
```diff
@@ -59,26 +59,15 @@ Target& Rover::RoverStatus::leftTarget()
...
-} // leftCacheTarget()
-
-Target& Rover::RoverStatus::rightCacheTarget() 
-{
-    return mCTargetRight;
-} // rightCacheTarget()
-
-RadioSignalStrength& Rover::RoverStatus::radio() 
-{
+Target& Rover::RoverStatus::target2() {
```
Comment (zwgold) [inline jetson/nav/rover.cpp:62]: Need to remove this, target2 no longer exists

File: jetson/nav/rover.cpp [hunk 2]
```diff
@@ -59,26 +59,15 @@ Target& Rover::RoverStatus::leftTarget()
...
+    return mTarget2;
+}
+RadioSignalStrength& Rover::RoverStatus::radio() {
     return mSignal;
 } // radio()
 
+
```
Comment (zwgold) [inline jetson/nav/rover.cpp:69]: Small spacing issues, try to remove to keep code condensed.

File: jetson/nav/rover.cpp (hunk 1 above)
Comment (cgiger00) [reply jetson/nav/rover.cpp:62]: will you remove it before the PR is merged in?

**Review by cgiger00 — COMMENTED** [review]
looks good to me, make the fixes zach requested plz

**Comment by zwgold** [comment]
Also merging in changes from main, do we want to do that so that we're up to date with main on this branch?

**Comment by mmartin4972** [comment]
> Also merging in changes from main, do we want to do that so that we're up to date with main on this branch?

Merging in changes from main is fine

**Comment by zwgold** [comment]
Closing this, Matt and I agreed we can add it in via his rebase of integration once it's approved in main, to make it easier.

**Pull request closed without merging**